drone/src/agent/docker.rs

```diff
@@ -284,16 +286,40 @@ impl DockerInterface {
         Ok((running, exit_code))
     }
 
-    pub async fn get_ip(&self, container_name: &str) -> Option<IpAddr> {
-        let inspect = self
-            .docker
-            .inspect_container(container_name, None)
-            .await
-            .ok()?;
+    pub async fn get_ip(&self, container_name: &str) -> Result<IpAddr> {
+        let inspect = self.docker.inspect_container(container_name, None).await?;
+
+        let network_settings = inspect
+            .network_settings
+            .ok_or_else(|| anyhow!("Inspect did not return network settings."))?;
+
+        if let Some(ip_addr) = network_settings.ip_address {
+            if !ip_addr.is_empty() {
+                return Ok(ip_addr.parse()?);
+            }
+        }
 
-        let port = inspect.network_settings?.ip_address?;
+        let networks = network_settings
+            .networks
+            .ok_or_else(|| anyhow!("Inspect did not return an IP or networks."))?;
+        if networks.len() != 1 {
+            return Err(anyhow!(
```

Comment:
so if Inspect does return an Ip, then having multiple networks does not throw an error. is that intended?

Reply:
The underlying assumption here is that if a container has an IP associated with it (meaning that it has an IP on the default bridge network), that IP should be accessible from the drone at that IP. Depending on the setup, this may require that the drone runs on the same network, but overall it seems like a reasonable default assumption.

If the container does not have an IP, we take a more conservative approach and use the IP associated with a network if and only if it's the _only_ network. We could go through this list and look for the “right” network, but if someone is attaching a container to multiple networks, there's various things they might be trying to do, none of which I've particularly thought through at this point, so rather than inferring their desired outcome I'd rather just give an error and have them discuss the use case.

Reply:
ye that makes sense!